linode/manager
# upcoming: [M3-8032] – Add Encrypted/Not Encrypted status to Node Pool table

## Description 📝
Add encryption status indication to the Node Pool table.

## Changes  🔄
- Add 3 new icons to `src/assets/icons`
- Destructure `disk_encryption` in `NodePoolsDisplay.tsx`, pass it down through to `NodeTable.tsx`
- Update some code patterns in `NodePool.tsx`

## Target release date 🗓️
5/28/24

## Preview 📷
| Encrypted  | Unencrypted   |
| ------- | ------- |
| ![Screenshot 2024-05-16 at 1 09 30 PM](https://github.com/linode/manager/assets/114682940/0155ff8e-fe6e-46c1-90e4-20ff75a0f09f) | ![Screenshot 2024-05-16 at 1 09 59 PM](https://github.com/linode/manager/assets/114682940/e8c4b761-4597-489b-9a68-1028a3496d66) |

## How to test 🧪
The easiest way to test would be to turn on the mocks and navigate to an LKE Details page, since the mocks I added cover the encrypted and unencrypted cases. Otherwise, you'd need your account in alpha to have a "legacy" cluster & pool to see the unencrypted variant.

### Verification steps
- Navigate to an LKE Details page with the devtools open. You should observe a GET request to `pools?page_size=200`
  - See the `disk_encryption` field for each pool. It should correspond with what's shown in the UI for each.
  - Confirm the Disk Encryption status looks good in mobile view also

## As an Author I have considered 🤔
